src/base/rss/rss_autodownloader.cpp

```diff
@@ -245,6 +245,7 @@ void AutoDownloader::processJob(const QSharedPointer<ProcessingJob> &job)
         }
 
         rule.setLastMatch(job->articleDate);
+        rule.matched(job->articleTitle);
```

Comment:
That method seems like it would be very easy for lastEpisodeName to be leaked at some point. And it's a lot of effort to go to, just to avoid a duplicate call to computeEpisodeName(), which only happens when there is a match.